internal/pkg/client/gitlab/fetch.go

```diff
@@ -0,0 +1,247 @@
+		return fmt.Errorf("%w. HTTP status code: %d", errFailedWebRequest, statusCode)
+	}
+
+	return nil
+}
+
+func (client *Client) createExport(projectName, branchName string) error {
+	project, response, err := client.Client.GetProject(projectName, nil)
+	if err != nil {
+		return fmt.Errorf("failed to get project info: %w", err)
+	}
+
+	if err := validateHTTPStatusCode(response.StatusCode); err != nil {
+		return err
+	}
+
+	branch, response, err := client.Client.GetBranch(project.ID, branchName)
+	if err != nil {
+		return fmt.Errorf("failed to get branch: %w", err)
+	}
+
+	if err := validateHTTPStatusCode(response.StatusCode); err != nil {
+		return err
+	}
+
+	commit, response, err := client.Client.GetCommit(project.ID, branch.Commit.ID, nil)
+	if err != nil {
+		return fmt.Errorf("failed to get commit info: %w", err)
+	}
+
+	if err := validateHTTPStatusCode(response.StatusCode); err != nil {
+		return err
+	}
+
+	// NOTE:
+	// If an authenticated user does not have permission to read_dependency,
+	// this request returns a 403 Forbidden status code.
+	export, response, err := client.Client.CreateDependencyListExport(commit.LastPipeline.ID, nil)
+	if err != nil {
+		return fmt.Errorf("failed to create dependency list export: %w", err)
+	}
+
+	if err := validateHTTPStatusCode(response.StatusCode); err != nil {
+		return err
+	}
+
+	if response.StatusCode == http.StatusForbidden {
+		return fmt.Errorf("%w", errForbiddenAccess)
+	}
+
+	client.Export = export
+
+	return nil
+}
+
+func (client *Client) pollExportUntilFinished() error {
+	const waitSeconds = 2
+	for !client.Export.HasFinished {
+		time.Sleep(waitSeconds * time.Second)
+
+		updatedExport, response, err := client.Client.GetDependencyListExport(client.Export.ID)
+		if err != nil {
+			return fmt.Errorf("failed to get dependency list export: %w", err)
+		}
+
+		if err := validateHTTPStatusCode(response.StatusCode); err != nil {
+			return err
+		}
+
+		client.Export = updatedExport
+	}
+
+	return nil
+}
+
+func (client *Client) downloadExport() ([]byte, error) {
+	sbomReader, response, err := client.Client.DownloadDependencyListExport(client.Export.ID)
+	if err != nil {
+		return nil, fmt.Errorf("failed to download dependency list: %w", err)
+	}
+
+	if err := validateHTTPStatusCode(response.StatusCode); err != nil {
+		return nil, err
+	}
+
+	var sbomData []byte
+
+	_, err = sbomReader.Read(sbomData)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read SBOM data: %w", err)
+	}
+
+	return sbomData, nil
+}
+
+func (client *Client) Fetch(fetchURL string, _ *bomctloptions.FetchOptions) ([]byte, error) {
+	url := client.Parse(fetchURL)
+	if url == nil {
+		return nil, fmt.Errorf("%w: %s", errInvalidGitLabURL, fetchURL)
+	}
+
+	domain := url.Hostname
+	if url.Port != "" {
+		domain = fmt.Sprintf("%s:%s", domain, url.Port)
+	}
+
+	baseURL := fmt.Sprintf("https://%s/api/v4", domain)
+	projectName := url.Path
+	branchName := url.Fragment
+
+	gitLabToken := os.Getenv("GITLAB_FETCH_TOKEN")
```

Comment:
Before documenting anything, I think it should be just `GITLAB_TOKEN`.

- it's more canonical and consistent with other CLI tools, GitLab's documentation, etc.
- it's reusable for other operations like push

I could also see an argument for prefixing it as `BOMCTL_GITLAB_TOKEN` in case we want to turn it into a command line option or something at some point.

Reply:
Agreed. I personally lean towards `BOMCTL_GITLAB_TOKEN`, since there is a possibility of a separate GitLab token with different permissions existing on the system which might occupy the variable `GITLAB_TOKEN`.

Reply:
I agree. I changed it to the recommended name: `BOMCTL_GITLAB_TOKEN`.

Where can I document that variable? I thought about the `bomctl fetch` help, but I know that we're trying to keep the command client-agnostic.

Reply:
As for location, I'd either put it in the fetch section in the README.md, or start an auth section instead. 

Would it make sense to make an issue to match this functionality in the other clients?